### ISSUE TITLE
[Feature]: Time labels appended inline in VS Code after custom method execution

### ISSUE THREAD

**Issue by guido-visser** [microsoft/playwright#33602]
### 🚀 Feature Request

When you run tests in VS Code, labels with the amount of milliseconds it took to execute a certain method will show:

`await locator.click(); - 102ms`

I have methods that execute certain parts of tests in a Page Object Model, but when that is executed, it doesn't show the execution time after that method was executed. It would be nice to have a way to add this functionality, so that I can see how long a certain method took to execute.

Current behavior:
![Image](https://github.com/user-attachments/assets/e271b8b6-22f6-4ea3-a768-6caabfaf35d7)

Wanted behavior:
![Image](https://github.com/user-attachments/assets/45ca22cc-30a1-4004-9179-df9a3d79cad4)

### Example

It would be great if Playwright just recognizes that it's waiting on a promise and will show how long it look to resolve that promise and shows the time in ms, like it does with native `.click()` or `expect` methods. However, I do recognize that it wouldn't be ideal for every promise, so this would be a proposal:

```
quicksearch = async (value: string) => {
       playwright.trackTiming();
       await page.getByTestId('search-btn').click();
       await page.getByTestId('search-field').fill(value);
}
```

Which would result in the wanted behavior image pictured above. This implementation would be very versatile and gives you the option to specify what method to keep track of in terms of timing.

### Motivation

It will give you a better overview of the timing of the tests that run in VS Code, including Page Object Model methods. It can also motivate a developer to make a more efficient method that will execute more efficiently, or maybe show bugs in code, because the execution takes longer than expected. This will show you that more easily.

**Comment by pavelfeldman** [comment]
Would be a nice feature, but I don't think we can do that unfortunately.

**Comment by guido-visser** [comment]
@pavelfeldman Thanks for replying! Bummer that it can't be done. A follow up question, though: Is it because it's the VS Code Extension that adds those labels, and since you can't provide a way to communicate with the extension, it's not possible? Could you expand on that? Just for some context for people that might stumble upon this feature request.
